### PR TITLE
feat: optout `new URL('./foo', import.meta.url)` import

### LINKED ISSUE #2745: [Feature Request]: A way to opt-out the `new URL('./foo', import.meta.url)` transform
### What problem does this feature solve?

A plugin may inject `new URL('./foo', import.meta.url)` that is not intended to be process by rolldown and will be processed by the plugin itself in `renderChunk` or other build hooks.
In this case, the plugin wants a way to opt-out the `new URL('./foo', import.meta.url)` transform.

Currently, rolldown outputs `Could not resolve, treating it as an external dependency` warning but there's no way to suppress this warning for that use case. `new URL` is generated on transform hook so the URL cannot be added to `external` option.

Also this would be useful for cases like `new URL('./non-existent-at-build', import.meta.url)` in deps (e.g. https://github.com/webpack/webpack/issues/16878).

### What does the proposed API look like?

Vite supports `/* @vite-ignore */` for `new URL('./foo', import.meta.url)` transform.

So maybe adding `/* @rolldown-ignore */`:
```js
new URL(/* @rolldown-ignore */ './foo', import.meta.url)
```

I'm using `new URL('./foo', '' + import.meta.url)` as a workaround for now.

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Closed #2745

### CODE REVIEW

File: crates/rolldown/tests/rolldown/topics/new_url/opt-out/main.js [hunk 1]
```diff
@@ -0,0 +1 @@
+/* @rolldown-ignore */ new URL('path-should-not-exists', import.meta.url);
```
Comment (hyf0) [inline crates/rolldown/tests/rolldown/topics/new_url/opt-out/main.js]:
```suggestion
new URL(/* @rolldown-ignore */ 'path-should-not-exists', import.meta.url);
```

What's the correct pattern @sapphi-red ?

**Review by hyf0 — APPROVED** [review]
LGTM. Make sure we don't support the wrong pattern.

**Comment by IWANABETHATGUY** [comment]
> LGTM. Make sure we don't support the wrong pattern.

Yeah, ref https://github.com/vitejs/vite/pull/16590/files

**Comment by IWANABETHATGUY** [comment]
* **#2755** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2755?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2754** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2754?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2753** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2753?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2753?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by IWANABETHATGUY** [comment]
merged directly and skip flaky test

**Pull request merged**